### DISCUSSION TITLE
Doubt about elf2hex

### DISCUSSION THREAD

**Discussion by claubersouza** [cpc/tce#266] [Q&A]
I'm studying how compilation for OpenAsip works. A big basic question, I don't understand why it uses elf2hex, I know it removes elf which is the link used in unix operating systems. But why does Risc-V need this?

**Comment by pjaaskel** [comment]
ELF is a program exchange/binary format it contains various sections with instructions and data which need to be loaded to the processor memory for execution. Hex is a dump of the program/data memory sections which should be loaded to the processor memory before boot. Since there is no operating system in the co-processor to perform "program loading" at runtime from the ELF, we dump the bit image and load the raw binary before booting (from a ROM or such) or from the OpenCL driver using a host CPU.

**Reply by claubersouza** [reply]
I was decoding the hex that was the output of elf2hex, I realized that the instructions don't make sense. From what I understand, there are other instructions. Would it be this ?

**Reply by claubersouza** [reply]
To be more objective, how can I decode hexdump?
